src/libopensc/card-myeid.c

```diff
@@ -768,11 +776,38 @@ myeid_convert_ec_signature(struct sc_context *ctx, size_t s_len, unsigned char *
 	unsigned char *buf;
 	size_t buflen;
 	int r;
-
-	assert(data && datalen);
-
-	if (*data != 0x30 || *(data + 1) != (datalen - 2) || *(data + 2) != 0x02)
+	size_t len_size = 1;
+	size_t sig_len = 0;
+
+	assert(data && datalen && datalen > 3);
+	
+	/*
+		more advanced validation is needed, because with 521 bit keys
+		length of the signature TLV is encoded in two bytes.
+		- Hannu Honkanen 6.11.2015
```

Comment:
It's not a TODO, it has been implemented already. Cleaned up and rewrote this comment.